liquibase-core/src/main/java/liquibase/snapshot/SnapshotGeneratorFactory.java

```diff
@@ -306,12 +306,13 @@ private void getContainerTypes(Class<? extends DatabaseObject> type, Database da
         SortedSet<SnapshotGenerator> generators = getGenerators(type, database);
 
         if ((generators != null) && !generators.isEmpty()) {
-            SnapshotGenerator generator = generators.iterator().next();
```

Comment:
I don't think this is a change we can make, because it breaks how the extension system is supposed to work.

The purpose of the getPriority() call and this spot that sorts the valid generators by that priority and then picks the single one with the highest priority is to allow extensions to replace the existing functionality and have the best implementation for the interface be used. 

For example, the standard ColumnSnapshotGenerator assumes a relational databases, and the MongodbColumnSnapshotGenerator knows how to work with mongo. Therefore this code needs to make sure that only the MongodbColumnSnapshot generator is used and NOT ColumnSnapshotGenerator.

Or even if ColumnSnapshotGenerator **_could_** work without error but OracleColumnSnapshotGenerator says  it has a higher priority, we need to just run the one implementation.

If you're thinking you need to execute all the implementations, it's probably pointing to a bug in one of the implementations which is either claiming it should run when it shouldn't, or not doing everything it should.

Reply:
This is the case. The fix is in the pro repo. Closing this PR.